WordPress/gutenberg
# Add storybook story for Disabled component

## Description
Adds a storybook story for the Disabled component as part of #17973

I had to make a decision on this story regarding what/how much to put in the component to show as disabled. I included all the "basic" form controls as well as a few of the more complex ones. I don't have a strong feeling on what should be shown, so if anyone has feedback on more/less child components just let me know.

## Concerns
I also ran into a few things with the storyshot integration I want to call out:
* It seems including another component in a new story increments an identifier used in the other stories:
![Comparing_master___add_disabled-component-story_·_WordPress_gutenberg_1](https://user-images.githubusercontent.com/6653970/75487414-5ed42680-597c-11ea-8a86-07830a0f6256.png) 
I am pretty sure this is intentional but didn't see it documented everywhere so I just wanted to double-check.
* The Disabled component uses a MutationObserver which jest doesn't support. Using the unit test for the Disabled component as a template I added a stub of window.MutationObserver in beforeAll() and removed it in afterAll(). This appears to work but doesn't seem ideal as none of the other storybook tests need it. If there is a better way to do this feel free to point me in that direction.

## How has this been tested?
* run `npm run storybook:dev`
* See the new Disabled story under components

## Screenshots
![Components___Disabled_-_Default_⋅_Storybook](https://user-images.githubusercontent.com/6653970/75488337-22092f00-597e-11ea-9004-a07b49aded5d.png)

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
